ldm/modules/lora_manager.py

```diff
@@ -0,0 +1,244 @@
+                        if child_module.__class__.__name__ == "Linear" or (child_module.__class__.__name__ == "Conv2d" and child_module.kernel_size == (1, 1)):
+                            # It's easier to just convert into "lora" naming instead of trying to inverse map from "lora" -> diffuser
+                            lora_name = prefix + '.' + name + '.' + child_name
+                            lora_name = lora_name.replace('.', '_')
+                            mapping[lora_name] = child_module
+                            self.hooks.append(child_module.register_forward_hook(self._make_hook(lora_name)))
+            return mapping
+
+        self.text_modules = find_modules(
+            LORA_PREFIX_TEXT_ENCODER, self.text_encoder, TEXT_ENCODER_TARGET_REPLACE_MODULE)
+        self.unet_modules = find_modules(
+            LORA_PREFIX_UNET, self.unet, UNET_TARGET_REPLACE_MODULE)
+
+    def _make_hook(self, layer: str):
+        def hook(module, input_h, output):
+            for lora in self.applied_loras.values():
+                lora_layer = lora.layers.get(layer, None)
+                if lora_layer is None:
+                    continue
+                output = output + \
+                    lora_layer.up(lora_layer.down(*input_h)) * \
+                    lora.multiplier * lora_layer.scale
+            return output
+        return hook
+
+    def _load_lora(self, name, path_file, multiplier=1.0):
+        lora = load_lora(name, path_file, self.device, self.dtype,
+                         self.text_modules, self.unet_modules, multiplier)
+        self.loras[name] = lora
+        return lora
+
+    def apply_lora_model(self, name, mult: float = 1.0):
+        path = Path(self.lora_path, name)
+        file = Path(path, "pytorch_lora_weights.bin")
+
+        if path.is_dir() and file.is_file():
+            print(f"loading lora: {path}")
+            self.unet.load_attn_procs(path.absolute().as_posix())
+        else:
+            # converting and saving in diffusers format
+            path_file = Path(self.lora_path, f'{name}.ckpt')
+            if Path(self.lora_path, f'{name}.safetensors').exists():
+                path_file = Path(self.lora_path, f'{name}.safetensors')
+
+            if not path_file.exists():
+                print(f">> Unable to find lora: {name}")
+                return
+
+            lora = self.loras.get(name, None)
+            if lora is None:
+                lora = self._load_lora(name, path_file, mult)
+
+            lora.multiplier = mult
+            self.applied_loras[name] = lora
+
+    def load_lora(self):
+        for name, data in self.loras_to_load.items():
+            self.apply_lora_model(name, data["mult"])
+
+        # unload any lora's not defined by loras_to_load
+        for name in list(self.loras.keys()):
+            if name not in self.loras_to_load:
+                self.unload_lora(name)
+
+    def unload_lora(self, lora_name: str):
```

Comment:
Are you still getting issues? This shouldn't be necessary. The hooks above on [L158](https://github.com/invoke-ai/InvokeAI/pull/2712/files#diff-e4627eddfb532a58110c2c00bdeec9802ce696916ea644627718c60d96e54b92R158) wouldn't do anything if no loras were selected in the prompt.

Reply:
The only problem I have found is prompting can be incorrect. It seems like A1111 works with [this lora](https://civitai.com/models/9421/natalie-portmanlora) with a simple prompt like:
```
<lora:nataliePortmanLora_v10:1> professional photograph of natlp, 4K, HDR
```

But, that doesn't seem to work for some reason. Instead, the prompt has to be tweaked:
```
<lora:nataliePortmanLora_v10:1> professional photograph of a woman, 4K, HDR
```

Reply:
Fix PR is out, didn't notice the ordering on the call into lora_manager prompt processing. It's always the little things...